src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -682,7 +829,78 @@ public void CopyTo(Array array, int index)
                     throw new ArgumentException(SR.Argument_InvalidOffLen);
                 }
 
-                for (int i = 0; i < m_length; i++)
+                int i = 0;
+
+                if (m_length < 32)
+                    goto LessThan32;
+
+                if (Avx2.IsSupported)
+                {
+                    Vector256<byte> shuffleMask;
+                    Vector256<byte> bitMask;
+                    Vector256<byte> ones = Vector256.Create((byte)1);
+                    fixed (byte* ptr = ShuffleMask_CopyToBoolArray)
+                    {
+                        shuffleMask = Avx.LoadVector256(ptr);
+                    }
+                    fixed (byte* ptr = BitMask_CopyToBoolArray)
+                    {
+                        bitMask = Avx.LoadVector256(ptr);
+                    }
+
+                    fixed (bool* destination = &boolArray[index])
+                    {
+                        for (; (i + Vector256<byte>.Count) <= m_length; i += Vector256<byte>.Count)
+                        {
+                            int bits = m_array[i / BitsPerInt32];
+                            Vector256<int> scalar = Vector256.Create(bits);
+                            Vector256<byte> shuffled = Avx2.Shuffle(scalar.AsByte(), shuffleMask);
+                            Vector256<byte> extracted = Avx2.And(shuffled, bitMask);
+
+                            // The extracted bits can be anywhere between 0 and 255, so we normalise the value to either 0 or 1
+                            // to ensure compatibility with "C# bool" (0 for false, 1 for true, rest undefined)
+                            Vector256<byte> normalized = Avx2.Min(extracted, ones);
```

Comment:
It seems you don't do this kind of normalization for `BitArray(bool[])` constructor

Reply:
This is handled by comparing the bytes with zero (checking if the bytes are false) then negating the result: https://github.com/dotnet/corefx/pull/41896/commits/72477e7611735af808f061018544d65b1f8b3ba2#diff-e2f01cf03382b7d63fc3a67ad77fcedcR140